src/util/settings.cpp

```diff
@@ -49,6 +55,62 @@ static void MergeSettings(const Settings& settings, const std::string& section,
 }
 } // namespace
 
+bool ReadSettings(const fs::path& path, std::map<std::string, SettingsValue>& values, std::vector<std::string>& errors)
+{
+    values.clear();
+    errors.clear();
+
+    fsbridge::ifstream file;
+    file.open(path);
+    if (!file.is_open()) return true; // Ok for file not to exist.
+
+    SettingsValue in;
+    if (!in.read(std::string{std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>()})) {
+        errors.emplace_back(strprintf("Unable to parse settings file %s", path.string()));
+        return false;
+    }
+
+    if (file.fail()) {
+        errors.emplace_back(strprintf("Failed reading settings file %s", path.string()));
+        return false;
+    }
+    file.close();
```

Comment:
why? Either this is redundant because this is already called when `file` goes out of scope, or it should be called before `file.fail()` (in cause you want to check for some kind of failure)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15935#discussion_r458619533

> why? Either this is redundant because this is already called when `file` goes out of scope, or it should be called before `file.fail()` (in cause you want to check for some kind of failure)

Just freeing a resource that's no longer needed, added comment